src/widget/wtimesignaturemenu.cpp

```diff
@@ -0,0 +1,82 @@
+#include "wtimesignaturemenu.h"
+
+#include <QtGui/QIntValidator>
+#include <QtWidgets/QHBoxLayout>
+
+namespace {
+constexpr int kMinBeatsPerBar = 2;
+constexpr int kMaxBeatsPerBar = 32;
+} // namespace
+
+WTimeSignatureMenu::WTimeSignatureMenu(QWidget* parent)
+        : QWidget(parent),
+          m_pBeatCountBox(make_parented<QSpinBox>(this)),
+          m_pBeatLengthBox(make_parented<QComboBox>(this)),
+          m_beat(mixxx::kInvalidFramePos) {
+    hide();
+    setWindowFlags(Qt::Popup);
+    setAttribute(Qt::WA_StyledBackground);
+    setObjectName("WTimeSignatureMenu");
+
+    QHBoxLayout* pMainLayout = new QHBoxLayout();
+    pMainLayout->addWidget(m_pBeatCountBox);
+    pMainLayout->addWidget(m_pBeatLengthBox);
+    setLayout(pMainLayout);
+    connect(m_pBeatCountBox,
+            QOverload<int>::of(&QSpinBox::valueChanged),
+            this,
+            &WTimeSignatureMenu::slotBeatCountChanged);
+
+    // 2 ^ index corresponds to beat size
+    m_pBeatLengthBox->addItem(tr("1"));
```

Comment:
```suggestion
    m_pBeatLengthBox->addItem("1");
```
No need to translate numbers